evm/chains/base.py

```diff
@@ -480,6 +488,26 @@ def from_genesis_header(cls, chaindb, genesis_header):
     #
     # Mining and Execution API
     #
+    def apply_transaction(self, transaction):
+        """
+        Applies the transaction to the current tip block.
+        """
+        vm = self.get_vm()
+        block, receipt, computation = vm.apply_transaction(transaction)
+        receipts = block.get_receipts(self.chaindb) + [receipt]
+
+        tx_root_hash, tx_kv_nodes = make_trie_root_and_nodes(block.transactions)
+        receipt_root_hash, receipt_kv_nodes = make_trie_root_and_nodes(receipts)
+        self.chaindb.persist_trie_data_dict(tx_kv_nodes)
+        self.chaindb.persist_trie_data_dict(receipt_kv_nodes)
```

Comment:
1. Just wondering, since now there's no "clean up function", after calling `Chain.apply_transaction(transaction)` (Maybe it's miner to select transactions or eth-tester), it seems the transactions and receipts will be in the database permanently...?
2. Those four lines appear twice, maybe add a simple function to deal with it?

Reply:
+1 though I'm going to be fixing this in a subsequent PR.

Reply:
Maybe add a big warning to the docstring here stating that this will generate/persist partial block state and that has a big performance hit?